### DISCUSSION TITLE
Restrict/Filter logs by namespace and level

### DISCUSSION THREAD

**Discussion by tiagobnobrega** [adzejs/adze#174] [Q&A] [answered]
Hi. Great work on this library. I have one question. Is there a way to define the least severe level that will be rendered for a specific namespace?

Let's say I have:

`logA = adze.namespace('nsA').seal();` and `logB = adze.namespace('nsB').seal();`. I want to render `Debug` or more severe logs from `logA`, but only `Warn` or more severe from `logB` and any other logs should only render `Error` or more severe messages.

What would be the best way to accomplish this?

**Comment by AJStacy** [comment, marked as answer]
Hey @tiagobnobrega, great question.

If you take a look at the documentation for the `seal()` modifier, you should see that it accepts a UserConfiguration value as a parameter (https://adzejs.com/reference/terminators.html#seal). Also, if you take a look at the UserConfiguration documentation, you should see that it accepts an `activeLevel` property (https://adzejs.com/reference/configuration.html#user-configuration).

So, instead of setting global configuration with the `setup()` function, rather provide specific configurations to each child logger in the `seal()` terminator.

```typescript
export const logA = adze.namespace('nsA').seal({ activeLevel: 'debug' });
export const logB = adze.namespace('nsB').seal({ activeLevel: 'warn' });
```

Keep in mind that the `setup()` function is meant for global overrides only. It's not required!

I hope this helps!

**Reply by AJStacy** [reply]
Also, I think this is a gap in the Getting Started documentation that I should explain in more detail.
